src/main/java/edu/harvard/iq/dataverse/DOIEZIdServiceBean.java

```diff
@@ -59,29 +59,54 @@ public DOIEZIdServiceBean() {
         } catch(Exception e){
             System.out.print("Other Error on ezidService.login(USERNAME, PASSWORD) - not EZIDException ");
         }
-    }    
-    
-    public String createIdentifier(Dataset dataset) {
-        String retString = "";
+    }
+
+    @Override
+    public boolean registerWhenPublished() {
+        return false;
+    }
+
+    @Override
+    public boolean alreadyExists(Dataset dataset) throws Exception {
+        logger.log(Level.FINE,"alreadyExists");
+        try {
+            HashMap<String, String> result = ezidService.getMetadata(getIdentifierFromDataset(dataset));
+            return result != null && !result.isEmpty();
+            // TODO just check for HTTP status code 200/404, sadly the status code is swept under the carpet
+        } catch (EZIDException e ){
+            logger.log(Level.WARNING, "alreadyExists failed");
+            logger.log(Level.WARNING, "String {0}", e.toString());
+            logger.log(Level.WARNING, "localized message {0}", e.getLocalizedMessage());
+            logger.log(Level.WARNING, "cause", e.getCause());
+            logger.log(Level.WARNING, "message {0}", e.getMessage());
+            throw e;
+        }
+    }
+
+    @Override
+    public String createIdentifier(Dataset dataset) throws Exception {
+        logger.log(Level.FINE,"createIdentifier");
         String identifier = getIdentifierFromDataset(dataset);
         HashMap<String, String> metadata = getMetadataFromStudyForCreateIndicator(dataset);
         metadata.put("_status", "reserved");
         try {
-            retString = ezidService.createIdentifier(identifier, metadata);
+            String retString = ezidService.createIdentifier(identifier, metadata);
             logger.log(Level.FINE, "create DOI identifier retString : " + retString);
+            return retString;
         } catch (EZIDException e) {
             logger.log(Level.WARNING, "Identifier not created: create failed");
             logger.log(Level.WARNING, "String {0}", e.toString());
             logger.log(Level.WARNING, "localized message {0}", e.getLocalizedMessage());
             logger.log(Level.WARNING, "cause", e.getCause());
             logger.log(Level.WARNING, "message {0}", e.getMessage());
-            return "Identifier not created "  +  e.getLocalizedMessage();
+            throw e;
```

Comment:
When comparing with DOIDataciteServiceBean, I see that one consistently logs INFO messages in stead of warnings. Is that intentional or a mistake?